src/metrics_server/connection_metrics.ts

```diff
@@ -93,38 +108,70 @@ export function isValidConnectionMetricsReport(
     return false;
   }
 
-  const MIN_BYTES_TRANSFERRED = 0;
-  const MAX_BYTES_TRANSFERRED = 1 * Math.pow(2, 40); // 1 TB.
-  for (const userReport of testObject.userReports) {
-    // Check that `bytesTransferred` is a number between min and max transfer limits
-    if (
-      typeof userReport.bytesTransferred !== 'number' ||
-      userReport.bytesTransferred < MIN_BYTES_TRANSFERRED ||
-      userReport.bytesTransferred > MAX_BYTES_TRANSFERRED
-    ) {
-      return false;
+  let i = testObject.userReports.length;
+  while (i--) {
+    const userReport = testObject.userReports[i];
+
+    // For backwards compatibility, we do not want to invalidate legacy report
+    // formats. Instead, we drop them silently.
+    if (!isCurrentUserReport(userReport)) {
+      testObject.userReports.splice(i, 1);
+      continue;
     }
 
-    if (
-      userReport.tunnelTimeSec &&
-      (typeof userReport.tunnelTimeSec !== 'number' || userReport.tunnelTimeSec < 0)
-    ) {
+    if (!isValidUserConnectionMetricsReport(userReport)) {
       return false;
     }
+  }
+
+  // Request is a valid HourlyConnectionMetricsReport.
+  return true;
+}
 
-    // We require at least 1 country to be set
-    const countries = userReport.countries ?? [];
-    if (countries.length === 0) {
+// Returns true iff testObject contains a valid HourlyUserConnectionMetricsReport.
+function isValidUserConnectionMetricsReport(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  testObject: any
+): testObject is HourlyUserConnectionMetricsReport {
+  if (!testObject) {
+    return false;
+  }
+
+  // Check that all required fields are present.
+  const requiredConnectionMetricsFields = ['countries', 'bytesTransferred'];
+  for (const fieldName of requiredConnectionMetricsFields) {
+    if (!testObject[fieldName]) {
       return false;
     }
-    // Check that all `countries` are strings.
-    for (const country of countries) {
-      if (typeof country !== 'string') {
-        return false;
-      }
+  }
+
+  // Check that `bytesTransferred` is a number between min and max transfer limits
+  if (
+    typeof testObject.bytesTransferred !== 'number' ||
+    testObject.bytesTransferred < 0 ||
+    testObject.bytesTransferred > TERABYTE
```

Comment:
The goal here is to try to prevent malicious reports. This was the max bytes for an access key. Then a location. Though we later realized that a server may use a single key, or all the traffic may be from a single location.

Still, we report hourly, and 1TB per hour is a lot (24 * 30 * 1TB a month). I doubt a server would reach that. Those that may are probably custom setups. 

Might be worth adding a comment.